tests/rptest/services/redpanda.py

```diff
@@ -1560,8 +1560,38 @@ def get_node_by_id(self, id):
                 return p
         return None
 
-    def stop_node(self, node, **kwargs):
-        pass
+    def stop_node(self,
+                  pod_name,
+                  signal=int(signal.SIGTERM),
+                  timeout=60,
+                  **kwargs):
+        """Stop a specific pod by name
+
+        Send a signal to the redpanda process running in k8s pod container.
+        The list of pod names can be found in self.pods property.
+
+        :param pod_name: string of pod name, e.g. 'rp-clo88krkqkrfamptsst0-5'
+        :param signal: int signal number
+        :param timeout: number of seconds to wait_until timeout
+        """
+
+        kill_cmd = f'kill -n {signal} 1'  # assume redpanda is pid 1 in the pod's container
+        # kubectl exec rp-clo88krkqkrfamptsst0-0 -n=redpanda -c=redpanda -- /bin/bash -c 'kill 1'
+        self._kubectl.exec(kill_cmd, pod_name=pod_name)
+
+        def pod_container_ready(pod_name):
+            # kubectl get pod rp-clo88krkqkrfamptsst0-0 -n=redpanda -o=jsonpath='{.status.containerStatuses[0].ready}'
+            return self._kubectl.cmd([
+                'get', 'pod', pod_name, '-n=redpanda',
+                "-o=jsonpath='{.status.containerStatuses[0].ready}'"
+            ]).decode()
+
```

Comment:
It would be nice to have a kind of self-check here that the pod actually restarted (and only once). Probably the easiest way is to check the "restarts" value for the pod before and after and ensure it has incremented by 1.

Reply:
i added assertion to check pod restartCount in dd9acd67ff9fc6c5d9fd4c7411435895f10671d1